test/functional/cmdLineTests/shareClassTests/SCCMLTests/ShareClassesCMLTests-1.xml

```diff
@@ -1049,6 +1051,120 @@
 		<output type="failure" caseSensitive="no" regex="no">corrupt</output>
 		<output type="failure" caseSensitive="yes" regex="no">Processing dump event</output>
 	</test>
+	
+	<test id="Test 57: Make sure that lambda expressions work and the classes get stored in the shared class cache" timeout="600" runPath=".">
+    	<command>$JAVA_EXE$ $currentMode$ $CP_HANOI$ $PROGRAM_LAMBDA$ 0</command>
+   		<output type="success" caseSensitive="yes" regex="no">Lambda test done!</output>
+    
+   		<output type="failure" caseSensitive="no" regex="no">Unhandled Exception</output>
+   		<output type="failure" caseSensitive="no" regex="no">corrupt</output>
+    	<output type="failure" caseSensitive="yes" regex="no">Processing dump event</output>
+	</test>
+
+	<test id="Test 58: Make sure that lambda classes are stored in the cache as orphans" timeout="600" runPath=".">
+	    <command>$JAVA_EXE$ $currentMode$,printstats=orphan</command>
+	    <output type="success" caseSensitive="yes" regex="yes" javaUtilPattern="yes">ORPHAN: org/openj9/test/lambdatests/Test1..Lambda.([\d]+)/0000000000000000 at</output>
+	    <output type="required" caseSensitive="yes" regex="yes" javaUtilPattern="yes">ORPHAN: org/openj9/test/lambdatests/Test1..Lambda.([\d]+)/0000000000000000 at</output>
+	    
+	    <output type="failure" caseSensitive="no" regex="no">Unhandled Exception</output>
+	    <output type="failure" caseSensitive="yes" regex="no">Exception:</output>
+	    <output type="failure" caseSensitive="no" regex="no">corrupt</output>
+	    <output type="failure" caseSensitive="yes" regex="no">Processing dump event</output>
+	</test>
+	
+	<test id="Test 59: Make sure that when the program runs again lambda classes are used from the cache and not stored again " timeout="600" runPath=".">
+	    <command>$JAVA_EXE$ $currentMode$ $lambdaMode$ $CP_HANOI$ $PROGRAM_LAMBDA$ 0</command>
+	    <output type="success" caseSensitive="yes" regex="no">Lambda test done!</output>
```

Comment:
Probably we can add a trace point that indicates the return value inside ROMClassBuilder::compareROMClassForEquality() and check that trace point here.